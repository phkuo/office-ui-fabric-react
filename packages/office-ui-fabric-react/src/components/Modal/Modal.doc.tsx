import * as React from 'react';
import { ModalBasicExample } from './examples/Modal.Basic.Example';

import { IDocPageProps } from '../../common/DocPage.types';
import { ModalStatus } from './Modal.checklist';

const ModalBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Modal/examples/Modal.Basic.Example.tsx') as string;
const ModalBasicExampleCodepen = require('!raw-loader!office-ui-fabric-react/lib/codepen/components/Modal/Modal.Basic.Example.Codepen.txt') as string;

export const ModalPageProps: IDocPageProps = {
  title: 'Modal',
  componentName: 'Modal',
  componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Modal',
  componentStatus: ModalStatus,
  examples: [
    {
      title: 'Modal',
      code: ModalBasicExampleCode,
      view: <ModalBasicExample />,
      codepenJS: ModalBasicExampleCodepen
    }
  ],
  propertiesTablesSources: [require<string>('!raw-loader!office-ui-fabric-react/src/components/Modal/Modal.types.ts')],
  overview: require<string>('!raw-loader!office-ui-fabric-react/src/components/Modal/docs/ModalOverview.md'),
  bestPractices: '',
  dos: require<string>('!raw-loader!office-ui-fabric-react/src/components/Modal/docs/ModalDos.md'),
  donts: require<string>('!raw-loader!office-ui-fabric-react/src/components/Modal/docs/ModalDonts.md'),
  isHeaderVisible: true,
  isFeedbackVisible: true
};
