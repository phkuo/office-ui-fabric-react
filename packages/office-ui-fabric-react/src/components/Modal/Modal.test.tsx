import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Modal } from './Modal';

describe('Modal', () => {
  it('renders Modal correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    const ReactDOM = require('react-dom');
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    const component = renderer.create(
      <Modal isOpen={true} className={'test-className'} containerClassName={'test-containerClassName'}>
        Test Content
      </Modal>
    );
    expect(component.toJSON()).toMatchSnapshot();

    ReactDOM.createPortal.mockClear();
  });
});
