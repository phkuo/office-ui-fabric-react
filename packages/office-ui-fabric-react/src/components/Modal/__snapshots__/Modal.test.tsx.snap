// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Modal correctly 1`] = `
<span
  className="ms-layer"
>
  <div
    className=
        ms-Fabric
        ms-Layer-content
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: #333333;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          visibility: visible;
        }
        & button {
          font-family: inherit;
        }
        & input {
          font-family: inherit;
        }
        & textarea {
          font-family: inherit;
        }
        button {
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          overflow: visible;
        }
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDoubleClick={[Function]}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onDragEnter={[Function]}
    onDragExit={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onSubmit={[Function]}
  >
    <div
      aria-modal="true"
      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "overflowY": undefined,
        }
      }
    >
      <div
        className=
            ms-Modal
            is-open
            test-className
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              align-items: center;
              background-color: transparent;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 100%;
              justify-content: center;
              opacity: 1;
              pointer-events: auto;
              position: fixed;
              transition: opacity 0.267s;
              width: 100%;
            }
      >
        <div
          className=
              ms-Overlay
              ms-Overlay--dark
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                background-color: rgba(0,0,0,.4);
                bottom: 0px;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                left: 0px;
                position: absolute;
                right: 0px;
                top: 0px;
              }
              @media screen and (-ms-high-contrast: active){& {
                border: 1px solid WindowText;
                opacity: 0;
              }
        />
        <div
          className=
              ms-Dialog-main
              test-containerClassName
              {
                background-color: #ffffff;
                box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.4);
                box-sizing: border-box;
                max-height: 100%;
                outline: 3px solid tranparent;
                overflow-y: auto;
                position: relative;
                text-align: left;
              }
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
        >
          <div
            className=
                ms-Modal-scrollableContent
                {
                  flex-grow: 1;
                  overflow-y: auto;
                }
          >
            Test Content
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
